src/core/libmaven/jsonReports.cpp

```diff
@@ -1,286 +1,678 @@
+
+
+JSONReports::JSONReports(MavenParameters* _mp, bool pollyUpload):
+    _uploadToPolly(pollyUpload)
+{
+    _mavenParameters=_mp;
+}
+
+
+void JSONReports::_writeGroup(PeakGroup& grp, ofstream& myfile) {
+    
+    //add labels to json file
+    char lab = grp.label;
+
+    PeakGroup* parentGroup = grp.getParent();
+    if (parentGroup) {
+        if (grp.label == '\0') {
+            lab = parentGroup->label;
+        }
+    }
+
+    myfile << setprecision(10);
+    myfile << "{\n";
+    myfile << "\"groupId\": " << grp.groupId ;
+
+    myfile << ",\n" << "\"label\": ";
+    myfile << "\"";
+    if (lab == 'g' || lab == 'b') myfile << lab;
+    myfile << "\"";
+
+    if(_uploadToPolly) {
+
+        int mlLabel =  (grp.markedGoodByCloudModel) ? 1 : (grp.markedBadByCloudModel) ? -1 : 0;
+        myfile  << ",\n" << "\"ml-label\": " << mlLabel;
+    }
+
+    myfile << ",\n" << "\"metaGroupId\": " << grp.metaGroupId ;
+    myfile << ",\n" << "\"meanMz\": " << grp.meanMz  ;
+    myfile << ",\n" << "\"meanRt\": " << grp.meanRt ;
+    myfile << ",\n" << "\"rtmin\": " << grp.minRt ;
+    myfile << ",\n" << "\"rtmax\": " << grp.maxRt ;
+    myfile << ",\n" << "\"maxQuality\": " << grp.maxQuality ;
+}
+
+
+void JSONReports::_writeCompoundLink(PeakGroup& grp,ofstream& myfile){
+    
+    myfile << setprecision(10);
+    double mz;
+    
+    int charge = _mavenParameters->getCharge(grp.getCompound());
+    mz = grp.getExpectedMz(charge);
+
+    if (mz == -1) {
+        mz = grp.meanMz;
+    }
+    
+    myfile << ",\n" << "\"compound\": { " ;
+
+    string compoundID = grp.getCompound()->id;
+    myfile << "\"compoundId\": "<< _sanitizeJSONstring(compoundID) ;
+    string compoundName = grp.getCompound()->name;
+    myfile << ",\n" << "\"compoundName\": "<< _sanitizeJSONstring(compoundName);
+    string formula = grp.getCompound()->formula;
+    myfile << ",\n" << "\"formula\": "<< _sanitizeJSONstring(formula);
+    myfile << ",\n" << "\"expectedRt\": " << grp.getCompound()->expectedRt;
+    myfile << ",\n" << "\"expectedMz\": " << mz ;
+    myfile << ",\n" << "\"srmID\": " << _sanitizeJSONstring(grp.srmId) ;
+    myfile << ",\n" << "\"tagString\": " << _sanitizeJSONstring(grp.tagString) ;
+
+    string fullTag = grp.srmId + grp.tagString;
+    string fullName=compoundName;
+    string fullID=compoundID;
+    if (fullTag.length()) {
+        fullName = compoundName + " [" + fullTag + "]";
+        fullID = compoundID + " [" + fullTag + "]";
+    }
+    myfile << ",\n" << "\"fullCompoundName\": "<< _sanitizeJSONstring(fullName);
+    myfile << ",\n" << "\"fullCompoundID\": "<< _sanitizeJSONstring(fullID) ;
+    myfile << "}" ; // compound
+}
+
+
+void JSONReports::_writePeak(PeakGroup& grp, ofstream& myfile, vector<mzSample*> vsamples){
+
+    myfile << setprecision(10);
+    
+    myfile << ",\n"<< "\"peaks\": [ " ;
+
+    for(std::vector<mzSample*>::iterator it = vsamples.begin(); it != vsamples.end(); ++it) {
+        if (it!=vsamples.begin()) {

```

Comment:
Please add, spaces between operators and comparators. You seem to ignore spaces a bit too much.